libbech32/bech32.cpp

```diff
@@ -481,11 +634,12 @@ bech32_error bech32_decode(bech32_HrpAndDp *output, char const *bstr, size_t bst
     if(hrpAndDp.hrp.empty() && hrpAndDp.dp.empty())
         return E_BECH32_INVALID_CHECKSUM;
 
-    if(hrpAndDp.hrp.size() > output->hrplen-1)
+    if(hrpAndDp.hrp.size() > output->hrplen)
```

Comment:
i see, so this wasn't a prior bug, but was a consequence of clarifying the semantics of hrplen.